### ISSUE TITLE
Allow overwriting the NavigationAppBar Layout

### ISSUE THREAD

**Issue by fischerscode** [bdlukaa/fluent_ui#708]
**Is your feature request related to a problem? Please describe.**
I want to implement some custom layout for the `NavigationAppBar`.

**Describe the solution you'd like**
It would be nice if `NavigationAppBar` would take an optional builder, in order to allow overwriting this behavior:
https://github.com/bdlukaa/fluent_ui/blob/0522bf08d28676619324c3ec5be5405fc93a22d2/lib/src/controls/navigation/navigation_view/view.dart#L889-L925

**Describe alternatives you've considered**
The `NavigationAppBar` could also have a build-like method instead of the builder. For building a custom design, one could extend `NavigationAppBar` and override the build method.

**Additional context**
I want the design shown in the screenshot.
The `MyApp` should never overlay the Hamburger Icon and the Close button should fill the corner.
Additionally there should be a `DragToMoveArea` stacked behind the Row.
![grafik](https://user-images.githubusercontent.com/45403027/215337108-80123db0-93de-467a-aa80-834a6e43accd.png)

**Comment by bdlukaa** [comment]
The `title` widget should take the whole app bar. `actions`, on the other hand, will be drawn on top of the `title`.

You can perform this layout by providing the desired layout to the `title` property

**Comment by fischerscode** [comment]
I think I tried this and it resulted in a white bar at the right side of the red button.

**Comment by fischerscode** [comment]
TLDR;
By using title and action and manually accounting for the actions width, my goal seems to be achievable, but I don't think it's an elegant solution and it still has a limitation.

### Using actions
![grafik](https://user-images.githubusercontent.com/45403027/216316626-f6609211-171c-4af1-b045-045f6e626bcb.png)
![grafik](https://user-images.githubusercontent.com/45403027/216316924-4b8269d5-be98-43a7-9f2a-d85c90dd8a0b.png)
The `CommandBarCard` is in front of the leading Button.
<details>
<summary>Code</summary>

```dart
import 'package:fluent_ui/fluent_ui.dart';
import 'package:flutter_acrylic/flutter_acrylic.dart' as flutter_acrylic;
import 'package:window_manager/window_manager.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await flutter_acrylic.Window.initialize();
  await WindowManager.instance.ensureInitialized();
  windowManager.waitUntilReadyToShow(
    const WindowOptions(
      minimumSize: Size(350, 600),
      center: true,
      titleBarStyle: TitleBarStyle.hidden,
      fullScreen: false,
    ),
    () async {
      await windowManager.show();
      await windowManager.setPreventClose(false);
      await windowManager.focus();
    },
  );

  runApp(FluentApp(
    debugShowCheckedModeBanner: false,
    home: NavigationView(
      appBar: NavigationAppBar(
          actions: Row(
        children: [
          Expanded(
              child: CommandBarCard(
            child: CommandBar(
                mainAxisAlignment: MainAxisAlignment.end,
                crossAxisAlignment: CrossAxisAlignment.center,
                primaryItems: [
                  CommandBarButton(onPressed: () {}, label: Text("Button1")),
                  CommandBarButton(onPressed: () {}, label: Text("Button2")),
                  CommandBarButton(onPressed: () {}, label: Text("Button3")),
                  CommandBarButton(onPressed: () {}, label: Text("Button4")),
                ]),
          )),
          SizedBox(
            width: 138,
            height: 50,
            child: WindowCaption(
              backgroundColor: Colors.transparent,
            ),
          ),
        ],
      )),
      pane: NavigationPane(items: []),
    ),
  ));
}
```
</details>

### Using title
![grafik](https://user-images.githubusercontent.com/45403027/216317471-6ccfd78c-ee15-42d3-b007-0a77efa0943a.png)
The red close button is not in the corner.
<details>
<summary>Code</summary>

```dart
import 'package:fluent_ui/fluent_ui.dart';
import 'package:flutter_acrylic/flutter_acrylic.dart' as flutter_acrylic;
import 'package:window_manager/window_manager.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await flutter_acrylic.Window.initialize();
  await WindowManager.instance.ensureInitialized();
  windowManager.waitUntilReadyToShow(
    const WindowOptions(
      minimumSize: Size(350, 600),
      center: true,
      titleBarStyle: TitleBarStyle.hidden,
      fullScreen: false,
    ),
    () async {
      await windowManager.show();
      await windowManager.setPreventClose(false);
      await windowManager.focus();
    },
  );

  runApp(FluentApp(
    debugShowCheckedModeBanner: false,
    home: NavigationView(
      appBar: NavigationAppBar(
          title: Row(
        children: [
          Expanded(
              child: CommandBarCard(
            child: CommandBar(
                mainAxisAlignment: MainAxisAlignment.end,
                crossAxisAlignment: CrossAxisAlignment.center,
                primaryItems: [
                  CommandBarButton(onPressed: () {}, label: Text("Button1")),
                  CommandBarButton(onPressed: () {}, label: Text("Button2")),
                  CommandBarButton(onPressed: () {}, label: Text("Button3")),
                  CommandBarButton(onPressed: () {}, label: Text("Button4")),
                ]),
          )),
          SizedBox(
            width: 138,
            height: 50,
            child: WindowCaption(
              backgroundColor: Colors.transparent,
            ),
          ),
        ],
      )),
      pane: NavigationPane(items: []),
    ),
  ));
}
```
</details>

### Using title and action
![grafik](https://user-images.githubusercontent.com/45403027/216318232-2e1c5a65-3b2d-4496-b092-772985789dd3.png)
Title and actions are stacked on top of each other.
<details>
<summary>Code</summary>

```dart
import 'package:fluent_ui/fluent_ui.dart';
import 'package:flutter_acrylic/flutter_acrylic.dart' as flutter_acrylic;
import 'package:window_manager/window_manager.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await flutter_acrylic.Window.initialize();
  await WindowManager.instance.ensureInitialized();
  windowManager.waitUntilReadyToShow(
    const WindowOptions(
      minimumSize: Size(350, 600),
      center: true,
      titleBarStyle: TitleBarStyle.hidden,
      fullScreen: false,
    ),
    () async {
      await windowManager.show();
      await windowManager.setPreventClose(false);
      await windowManager.focus();
    },
  );

  runApp(FluentApp(
    debugShowCheckedModeBanner: false,
    home: NavigationView(
      appBar: NavigationAppBar(
        title: Expanded(
            child: CommandBarCard(
          child: CommandBar(
              mainAxisAlignment: MainAxisAlignment.end,
              crossAxisAlignment: CrossAxisAlignment.center,
              primaryItems: [
                CommandBarButton(onPressed: () {}, label: Text("Button1")),
                CommandBarButton(onPressed: () {}, label: Text("Button2")),
                CommandBarButton(onPressed: () {}, label: Text("Button3")),
                CommandBarButton(onPressed: () {}, label: Text("Button4")),
              ]),
        )),
        actions: SizedBox(
          width: 138,
          height: 50,
          child: WindowCaption(
            backgroundColor: Colors.transparent,
          ),
        ),
      ),
      pane: NavigationPane(items: []),
    ),
  ));
}
```
</details>

### Using title and action and manually accounting for the actions width
![grafik](https://user-images.githubusercontent.com/45403027/216318917-b0e00cf9-1ba0-4a47-9f70-5f3f4141285c.png)
Actually works, I did not think about this solution earlier, but I don't think this solution is elegant.
<details>
<summary>Code</summary>

```dart
import 'package:fluent_ui/fluent_ui.dart';
import 'package:flutter_acrylic/flutter_acrylic.dart' as flutter_acrylic;
import 'package:window_manager/window_manager.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await flutter_acrylic.Window.initialize();
  await WindowManager.instance.ensureInitialized();
  windowManager.waitUntilReadyToShow(
    const WindowOptions(
      minimumSize: Size(350, 600),
      center: true,
      titleBarStyle: TitleBarStyle.hidden,
      fullScreen: false,
    ),
    () async {
      await windowManager.show();
      await windowManager.setPreventClose(false);
      await windowManager.focus();
    },
  );

  runApp(FluentApp(
    debugShowCheckedModeBanner: false,
    home: NavigationView(
      appBar: NavigationAppBar(
        title: Row(
          children: [
            Expanded(
                child: CommandBarCard(
              child: CommandBar(
                  mainAxisAlignment: MainAxisAlignment.end,
                  crossAxisAlignment: CrossAxisAlignment.center,
                  primaryItems: [
                    CommandBarButton(onPressed: () {}, label: Text("Button1")),
                    CommandBarButton(onPressed: () {}, label: Text("Button2")),
                    CommandBarButton(onPressed: () {}, label: Text("Button3")),
                    CommandBarButton(onPressed: () {}, label: Text("Button4")),
                  ]),
            )),
            SizedBox(width: 138),
          ],
        ),
        actions: SizedBox(
          width: 138,
          height: 50,
          child: WindowCaption(
            backgroundColor: Colors.transparent,
          ),
        ),
      ),
      pane: NavigationPane(items: []),
    ),
  ));
}
```
</details>
Disadvantage about this solution:
It's not possible to add a `DragToMoveArea` that triggers when the user drags at the marked area. This would be possible with #709.

![grafik](https://user-images.githubusercontent.com/45403027/216320695-2699d325-a1b7-4316-b7d8-9908b251660a.png)

**Comment by bdlukaa** [comment]
On the native side, the `title` always go behind the window features (title and actions). But I also agree that there is a "search bar" (or something related) in some native UWP apps

This is a hard problem to solve, because desktop aims adaptability of different window sizes, otherwise it can cause to some weird effects. Here is a screenshot of "Slack" with a small size:

![image](https://github.com/bdlukaa/fluent_ui/assets/45696119/974a56f1-143c-4353-ba13-855239c3f9de)

On the other hand, on Task Manager, the "search bar" window feature works differently. If there isn't enough space, the bar is changed to a button:

![Before / Small](https://github.com/bdlukaa/fluent_ui/assets/45696119/1356d242-8055-49b8-ac83-936c442d4aec)

![After / Big](https://github.com/bdlukaa/fluent_ui/assets/45696119/b90ece83-14d0-47d7-93dd-864d0950d213)


But, after it all, none of the above is described in the Win UI documentation. In fact, the only option Microsoft provides to add some custom content is either:
![image](https://github.com/bdlukaa/fluent_ui/assets/45696119/4f7bbc22-1f25-4a30-90f1-59414a721c69)

**Comment by bdlukaa** [comment]
A better option would be making `NavigationAppBar` a widget, so that its `build` function could be overwritten.
